src/Sarif.Viewer.VisualStudio/Views/CallTreeView.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Copyright (c) Microsoft. All rights reserved. 
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+
+namespace Microsoft.Sarif.Viewer.Views
+{
+    using System;
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using System.Windows;
+    using System.Windows.Controls;
+    using Microsoft.Sarif.Viewer.Models;
+
+    /// <summary>
+    /// Implementation of a tree view for the call tree.
+    /// </summary>
+    /// <remarks>
+    /// The selection in the tree view is maintained by the data model through the <see cref="CallTree.SelectedItem"/> property.
+    /// What this class is ensuring that when the selected item property is modified in the data model 
+    /// that the "parent" UI elements (tree view items) in the tree view are properly expanded so the element is visible.
+    /// The class must also perform the reverse logic of when the selection changes in the tree view ensure that it is reflected
+    /// back in the data model.
+    /// A future enhancement would be to separate out the "view logic data" from the "SARIF data model". Specifically the concept
+    /// of the selected item in the view should be distinct and separate from <see cref="CallTree.SelectedItem"/> as that is not
+    /// a "SARIF" concept.
+    /// </remarks>
+    internal class CallTreeView : TreeView
+    {
+        public CallTreeView()
+        {
+            SelectedItemChanged += CallTree_TreeView_SelectionChanged;
+            Loaded += CallTree_TreeView_Loaded;
+            Unloaded += CallTree_TreeView_Unloaded;
+        }
+
+        protected override void OnInitialized(EventArgs e)
+        {
+            base.OnInitialized(e);
+
+            // Subscribe to the call tree's property changed notification
+            // so that this control can properly handle any changes to the selected item
+            // (selected call tree node).
+            if (DataContext is CallTree callTree &&
+                callTree is INotifyPropertyChanged notifyPropertyChanged)
+            {
+                notifyPropertyChanged.PropertyChanged += CallTree_PropertyChanged;
+            }
+        }
+
+        /// <summary>
+        /// Called when properties of the <see cref="CallTree"/> is modified .
```

Comment:
>  [](start = 76, length = 1)

remove space

Reply:
and "are modified"

---
In reply to: [503387456](https://github.com/microsoft/sarif-visualstudio-extension/pull/214#discussion_r503387456) [](ancestors = 503387456)